src/backend.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) Nitrokey GmbH
+// SPDX-License-Identifier: Apache-2.0 or MIT
+
+mod data;
+
+use trussed::{
+    backend::Backend,
+    error::Result,
+    platform::Platform,
+    serde_extensions::ExtensionImpl,
+    service::ServiceResources,
+    store::filestore::Filestore,
+    types::{CoreContext, Location, PathBuf},
+};
+
+use crate::{
+    extension::{reply, AuthExtension, AuthReply, AuthRequest},
+    PIN_PATH,
+};
+use data::PinData;
+
+/// A basic implementation of the [`AuthExtension`][].
+///
+/// This implementation stores PINs together with their retry counters on the filesystem.  PINs are
+/// hashed with SHA-256 using a salt that is generated per PIN.
+#[derive(Clone, Debug)]
+pub struct AuthBackend {
+    location: Location,
+}
+
+impl AuthBackend {
+    /// Creates a new `AuthBackend` using the given storage location for the PINs.
+    pub fn new(location: Location) -> Self {
+        Self { location }
```

Comment:
I also thought about that.  `trussed` is already used for the Trussed device key and certificate.  I’d prefer something like `ext/auth`.  I did not use that yet because I wasn’t sure how well littlefs2 would handle it.  We would end up with `<client>/ext/auth/pin/00`, so two additional directory levels.

Reply:
Why is the `pin` directory necessary?

Reply:
In the future, we might also want to store policies in the file system.  I think it would make sense to separate the different data types from the beginning.  Then we can e. g. use key IDs as file names for key policies.

Reply:
True.  Having a separate directory is a bit more convenient (easier deletion of all files, easier path construction), but using distinct filename patterns should work too.

Regarding a dedicated subdirectory for the extension, we should actually have directories for the backends instead.  Multiple backends can implement the same extension using different data structures.